msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfiguration.kt

```diff
@@ -0,0 +1,214 @@
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client
+
+import android.text.TextUtils
+import com.google.gson.annotations.SerializedName
+import com.microsoft.identity.client.configuration.AccountMode
+import com.microsoft.identity.client.exception.MsalClientException
+import com.microsoft.identity.common.java.authorities.CIAMAuthority
+import com.microsoft.identity.common.java.authorities.NativeAuthCIAMAuthority
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.logging.Logger
+import com.microsoft.identity.common.java.providers.nativeauth.NativeAuthConstants
+import lombok.Getter
+import lombok.experimental.Accessors
+import java.io.Serializable
+
+/**
+ * Extends [PublicClientApplicationConfiguration] to add a few extra fields specific to Native Auth
+ * app implementation.
+ */
+@Accessors(prefix = ["m"])
+@Getter
+class NativeAuthPublicClientApplicationConfiguration :
+    PublicClientApplicationConfiguration(),
+    Serializable {
+    companion object {
+        private val TAG = NativeAuthPublicClientApplicationConfiguration::class.java.simpleName
+        private val VALID_CHALLENGE_TYPES = listOf(NativeAuthConstants.GrantType.PASSWORD,
+            NativeAuthConstants.GrantType.OOB, NativeAuthConstants.GrantType.REDIRECT)
+    }
+
+    private object NativeAuthSerializedNames {
+        const val CHALLENGE_TYPES = "challenge_types"
+        const val USE_REAL_AUTHORITY = "use_real_authority"
+        const val DC = "dc"
+    }
+
+    //List of challenge types supported by the client.
+    //For a complete list of challenge types see [NativeAuthConstants.GrantType]
+    @SerializedName(NativeAuthSerializedNames.CHALLENGE_TYPES)
+    private var challengeTypes: List<String>? = null
+
+    //The mock API authority used for testing will be rejected by validation logic run on
+    // instantiation. This flag is used to bypass those checks in various points in the application
+    @SerializedName(NativeAuthSerializedNames.USE_REAL_AUTHORITY)
+    var useRealAuthority: Boolean? = null
```

Comment:
I don't think this should be exposed in the configuration here.

Reply:
If you are already using BuildConfig then you certainly don't need it on the configuration.

Reply:
This is a flag that should be settable by the calling application (e.g. our sample app makes use of the mock API flows), and AFAIK there is no way for a calling app to change the library's BuildConfig